core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -202,6 +202,10 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
     eventManager.put(UncleanLeaderElectionEnable)
   }
 
+  private[kafka] def enableTopicUncleanLeaderElection(topic: String): Unit = {
+    eventManager.put(TopicUncleanLeaderElectionEnable(topic))
```

Comment:
We probably only want to enqueue the event if the controller is active.

Reply:
We have a check [here](https://github.com/apache/kafka/pull/4957/files#diff-ed90e8ecc5439a5ede5e362255d11be1R1037). This is similar to other ControllerEvent implementations.

Reply:
Thanks for the explanation.   For now, I have a added check for the new event. Please take a look.

Reply:
Looking at the code again, it seems that the controller event manager is started on every broker. So, what you had before is probably fine too. Adding this additional check is probably slightly better.